kernel/proc/scheduler.md

```diff
@@ -0,0 +1,31 @@
+# Kernel - Processes and threads - Scheduler
+The operating system scheduler controls threads execution with a predetermined policy. It is a part of Phoenix-RTOS having the most significant influence on the performance and responsiveness of the whole system.
+
+## Scheduling policy
+The scheduling algorithm is defined in the `threads_schedule` function. It is invoked by timer interrupt or voluntary reschedule (`hal_cpuReschedule`). Each function's invocation changes the context of the currently executed thread to the next one, available in a thread's ready queue. The scheduling policy determined which thread shall be chosen for the next execution. In Phoenix-RTOS, there is used a priority preempted round-robin algorithm. There are eight priority levels, the smallest value holds the highest priority.
+
+
+The thread management unit contains an eight-element array, where each of them holds a pointer to the list of threads of the same priority. A scheduling algorithm is defined as follows:
+
+1. The spinlock within thread management unit is set before any operations on common data.
```

Comment:
Maybe just say `threads_common.spinlock` is set? I don't know how technical we want to be in this documentation, there're already references to functions, so I guess it would be ok